src/index.js

```diff
@@ -102,39 +132,47 @@ class Dropzone extends React.Component {
 
   onDrop(e) {
     const { onDrop, onDropAccepted, onDropRejected, multiple, disablePreview } = this.props;
-    const fileList = getDataTransferItems(e, multiple);
     const acceptedFiles = [];
     const rejectedFiles = [];
 
     // Stop default browser behavior
     e.preventDefault();
 
-    // Reset the counter along with the drag on a drop.
-    this.enterCounter = 0;
+    // Reset the target list along with the drag on a drop.
+    this.dragTargets = [];
     this.isFileDialogActive = false;
 
-    fileList.forEach((file) => {
-      if (!disablePreview) {
-        file.preview = window.URL.createObjectURL(file); // eslint-disable-line no-param-reassign
-      }
+    // Don't fire callbacks if a child Dropzone component has already handled this event
+    // eslint-disable-next-line no-underscore-dangle
+    if (!e.__dropzoneChildHandledThis) {
```

Comment:
The problem is that the parent needs to know whether some child has already fired an onDrop() callback. If a child has fired, reset the dragActive/dragReject/dragTargets state and don't fire onDrop(). If a child has not fired, reset the state and do fire onDrop().

If we used `event.stopPropagation()`, the parent would never receive the event, so its state would remain active, and the render of the parent would still be styled with `dragActive=true` after a drop on the child.

The order is like:
1: Document `drop` handler (if installed via preventGlobalDrops=true)
2: Child `drop` handler
3: Parent `drop` handler